test/listvolume_tester.py

```diff
@@ -0,0 +1,40 @@
+# import mock
+import test.fake_3par_data as data
+from hpedockerplugin import exception as hpe_exc
+import test.hpe_docker_unit_test as hpedockerunittest
+from hpe3parclient import exceptions
+from oslo_config import cfg
+CONF = cfg.CONF
+
+
+class ListVolumeUnitTest(hpedockerunittest.HpeDockerUnitTestExecutor):
+    def _get_plugin_api(self):
+        return 'volumedriver_list'
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+    def override_configuration(self, config):
+        pass
+
+    # TODO: check_response and setup_mock_objects can be implemented
+    # here for the normal happy path TCs here as they are same
+
+
+class TestListVolumeDefault(ListVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": '', 'Volumes':[]})

```

Comment:
Response will have list of volumes. So comparison can be made to what is received vs what was set to return from mock_etcd.get_all_vols()

Reply:
Depends on the first mentioned comment. ["Tried it. But I dont have the idea what is the expected out for get_all_vols call.. This is making it fail.. Would be great if you can share an example response."]